tests/test_api.py

```diff
@@ -730,3 +732,50 @@ def test_delete_build_auth(testclient):
 
     # r = schema.APIResponse.parse_obj(response.json())
     # assert r.status == schema.APIStatus.ERROR
+
+
+def test_api_cancel_build(testclient):
+    build_id = 1
+
+    testclient.login()
+    response = testclient.put(f"api/v1/build/{build_id}")
+    r = schema.APIPostSpecification.parse_obj(response.json())
+    assert r.status == schema.APIStatus.OK
+
+    new_build_id = r.data.build_id
+
+    # delay to ensure the build kicks off
+    build_timeout = 180
+    building = False
+    start = time.time()
+    while time.time() - start < build_timeout:
+        response = testclient.get(f"api/v1/build/{new_build_id}")
+        response.raise_for_status()
+
+        r = schema.APIGetBuild.parse_obj(response.json())
```

Comment:
This should also have the standard `assert r.status == schema.APIStatus.OK` to fail early if there are problems. It will fail due to building not being True, but that'd be harder to debug later.